linode/manager
# M3 1697 refector storing tokens

**Purpose:**
Instead of grabbing the oauth token from browser local storage on every login just have a preconfigured MANAGER_OAUTH token in the .env file. Added addition logic to retrieve token from browser local storage if an oauth token was not provided, to maintain current jenkins runs. An oauth token called automation has been created for prod-test-001 - 012 and stored in the notes field in 1Password. 

**To Test:**
1st Scenario - A single user no preconfigured oauth - just have the MANAGER_USER and MANGER_PASS configured in the .env file. Place a break point in any spec file by adding a browser.debug(); in an it block. Run:
yarn && yarn start
yarn selenium
yarn e2e --file {project relative spec file with debug line}

Once the break point is reached in open a terminal cd into manger directory and cat ~/e2e/creds.js. Ensure that the 'token' field is not empty and the 'isPresetToken' field is false.

2nd Scenario -   A single user preconfigured oauth -have the MANAGER_USER, MANGER_PASS, and MANAGER_OAUTH configured in the .env file. Place a break point in any spec file by adding a browser.debug(); in an it block. Run:
yarn && yarn start
yarn selenium
yarn e2e --file {project relative spec file with debug line}

Once the break point is reached in open a terminal cd into manger directory and cat ~/e2e/creds.js. Ensure that the 'token' field is not empty and the 'isPresetToken' field is true.

3rd Scenario -Two users no preconfigured oauth - have the MANAGER_USER, MANGER_PASS, MANAGER_USER_2, and MANAGER_PASS_2 configured in the .env file. Place a break point in any spec file by adding a browser.debug(); in an it block. Run:
yarn && yarn start
yarn selenium
yarn e2e --spec={project relative spec file with debug line},{project relative path to a different spec file}

Once the break point is reached in open a terminal cd into manger directory and cat ~/e2e/creds.js. Ensure that both the 'token' fields are not empty and both the 'isPresetToken' fields are false.

4th Scenario -   Two users with preconfigured oauth -have the MANAGER_USER, MANGER_PASS, MANAGER_OAUTH, MANAGER_USER_2, MANGER_PASS_2, MANAGER_OAUTH_2 configured in the .env file. Place a break point in any spec file by adding a browser.debug(); in an it block. Run:
yarn && yarn start
yarn selenium
yarn e2e --spec={project relative spec file with debug line},{project relative path to a different spec file}

Once the break point is reached in open a terminal cd into manger directory and cat ~/e2e/creds.js. Ensure that both the 'token' fields are not empty and both the 'isPresetToken' fields are true.
